### PR TITLE
prometheus: Initialize all metrics

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

## Tips:

* commit message titles must be in the form:
```topic: Capitalized message with no trailing period```
or:
```topic, topic2: Capitalized message with no trailing period```

* golang code must be formatted according to the standard, please run:
```
make gofmt		# formats the entire project correctly
```
or format a single golang file correctly:
```
gofmt -w yourcode.go
```

* please rebase your patch against current git master:
```
git checkout master
git pull origin master
git checkout your-feature
git rebase master
git push your-remote your-feature
hub pull-request	# or submit with the github web ui
```

* after a patch review, please ping @purpleidea so we know to re-review:
```
# make changes based on reviews...
git add -p		# add new changes
git commit --amend	# combine with existing commit
git push your-remote your-feature -f
# now ping @purpleidea in the github PR since it doesn't notify us automatically
```

### CODE REVIEW

File: prometheus/prometheus.go [hunk 1]
```diff
@@ -170,7 +170,18 @@ func (obj *Prometheus) InitKindMetrics(kinds []string) error {
...
+
+		failures := []string{"soft", "hard"}
+		for _, f := range failures {
+			failLabels := prometheus.Labels{"kind": kind, "failure": f}
+			obj.failedResourcesTotal.With(failLabels)
+			obj.failedResources.With(failLabels)
+
+		}
 	}
+
```
Comment (purpleidea) [inline prometheus/prometheus.go]: why this nl?

File: prometheus/prometheus_test.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package prometheus
```
Comment (purpleidea) [inline prometheus/prometheus_test.go]: file needs standard header or tests will fail.

File: prometheus/prometheus_test.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+func TestInitKindMetrics(t *testing.T) {
+	var prom Prometheus
+	prom.Init()
+
+	prom.InitKindMetrics([]string{"file", "exec"})
+
+	gatherer := prometheus.DefaultGatherer
+	metrics, err := gatherer.Gather()
+	if err != nil {
+		t.Errorf("Error while gathering resources: %s", err)
```
Comment (purpleidea) [inline prometheus/prometheus_test.go]: do you want to  `return` after this Errorf happens or continue the test?

File: prometheus/prometheus_test.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+			2, 0,
+		},
+	}
+
+	for _, metric := range metrics {
+		for name, count := range expectedMetrics {
+			if *metric.Name == name {
+				value := len(metric.Metric)
+				expectedMetrics[name] = [2]int{count[0], value}
+				fmt.Printf("Found %d\n", count[1])
```
Comment (purpleidea) [inline prometheus/prometheus_test.go]: t.Logf instead please

Comment (roidelapluie) [reply prometheus/prometheus_test.go]: that was debug

**Pull request merged**